add_new_data_sources.py

```
@@ -0,0 +1,245 @@
+import json
+import os
+import sys
+import time
+import logging
+from senzing import G2ConfigMgr, G2Exception, G2Config
+
+# init vars
+module_name = 'pyG2'       
+verbose_logging = False 
+# sys path
+python_path = "{0}/python".format(
+    os.environ.get("SENZING_G2_DIR", "/opt/senzing/g2"))
+sys.path.append(python_path)
+
+# init senzing
+def init_senzing(log):
+    # paths
+    data_dir = os.environ.get("SENZING_DATA_DIR", "/opt/senzing/data")
+    etc_dir = os.environ.get("SENZING_ETC_DIR", "/etc/opt/senzing")
+    g2_dir = os.environ.get("SENZING_G2_DIR", "/opt/senzing/g2")
+
+    # create senzing config
+    config_path = etc_dir
+    support_path = os.environ.get("SENZING_DATA_VERSION_DIR", data_dir)
+    resource_path = "{0}/resources".format(g2_dir)
+
+    sql_connection = os.environ.get(
+        "SENZING_SQL_CONNECTION", "sqlite3://na:na@/var/opt/senzing/sqlite/G2C.db")
+
+    senzing_config_dictionary = {
+        "PIPELINE": {
+            "CONFIGPATH": config_path,
+            "SUPPORTPATH": support_path,
+            "RESOURCEPATH": resource_path
+        },
+        "SQL": {
+            "CONNECTION": sql_connection,
+        }
+    }
+
+    senzing_config_json = json.dumps(senzing_config_dictionary)
+
+    '''# init vars
+    module_name = 'pyG2'
+    verbose_logging = False'''
+
+    # init a G2ConfigMgr instance
+    
+    g2_configuration_manager = G2ConfigMgr()
+    try:
+        g2_configuration_manager.init(
+            module_name,
+            senzing_config_json,
+            verbose_logging)
+
+    except G2Exception.G2ModuleGenericException as err:
+        log.info(' %s' % err)

```

As a general comment in this code, you really need to raise the exception so the process stops.  Right now you are logging and blindly continuing which is likely to lead to failure... and your overall process may instead return success.

---

Done this via sys.exit(1)